ext/exif/exif.c

```diff
@@ -3405,6 +3405,15 @@ static int exif_process_IFD_TAG(image_info_type *ImageInfo, char *dir_entry, cha
 				break;
 
 			case TAG_USERCOMMENT:
+				if (ImageInfo->UserComment) {
+					EFREE_IF(ImageInfo->UserComment);
```

Comment:
These `EFREE_IF`s should just be `efree`s, as you already do the if check yourself. Or alternatively, if you would like to use `EFREE_IF` you can drop the outer ifs and do something like this:

```c
EFREE_IF(ImageInfo->UserComment);
ImageInfo->UserComment = NULL;
EFREE_IF(ImageInfo->UserCommentEncoding);
ImageInfo->UserCommentEncoding = NULL;
```

Reply:
Oddly enough, that's how I started, but I think I was noticing some problems, so I switched to the `if` statements. The problems may have been because I wasn't setting to the values to `NULL` originally. I'll go back and try this, since I think it's cleaner.